### PR TITLE
feat: add openziti to core contracts

### PR DESCRIPTION
**Pull request by dovholuknf**
Uses the authInjector to set the RoundTripper to use when making requests. Necessary to integrate with OpenZiti. See go-mod-bootstrapper at https://github.com/edgexfoundry/go-mod-bootstrap

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x ] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x ] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?)
- [ ] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

## Testing Instructions
This is a sweeping change that crosses numerous repos. I don't know how to effectively answer this particular question and this one change is insufficient for testing. A larger, cross-cutting PR/effort is needed to put all the pieces together. I could use help with knowing how to satisfy this

## New Dependency Instructions (If applicable)
none

### CODE REVIEW

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Review by cloudxxx8 — CHANGES_REQUESTED** [review]
Please help make sure the unit test are all passed

[2024-02-01T03:30:52.756Z] ?   	github.com/edgexfoundry/go-mod-core-contracts/v3/clients	[no test files]
[2024-02-01T03:30:58.022Z] # github.com/edgexfoundry/go-mod-core-contracts/v3/clients/http [github.com/edgexfoundry/go-mod-core-contracts/v3/clients/http.test]
[2024-02-01T03:30:58.022Z] clients/http/authinjector_test.go:19:9: cannot use &emptyAuthenticationInjector{} (value of type *emptyAuthenticationInjector) as interfaces.AuthenticationInjector value in return statement: *emptyAuthenticationInjector does not implement interfaces.AuthenticationInjector (missing method RoundTripper)
[2024-02-01T03:30:58.022Z] FAIL	github.com/edgexfoundry/go-mod-core-contracts/v3/clients/http [build failed]

**Review by lenny-intel — APPROVED** [review]
LGTM once unit test is fixed

**Comment by dovholuknf** [comment]
The interface was not satisfied as I'd changed it. I've pushed a fix out that should hopefully fix it

**Review by lenny-intel — APPROVED** [review]
LGTM

File: clients/doc_test.go
Comment (cloudxxx8) [inline clients/doc_test.go:1]: why do we need this file?

Comment (dovholuknf) [reply clients/doc_test.go:1]: geesh - that should not have been committed. i'll remove it! thx (i was trying to get goland to run the tests and put that file there as a marker)

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Pull request merged**